contracts/peg/curve/CurveRouter.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {ICurveRouter} from "./ICurveRouter.sol";
+import {ICurvePool} from "./ICurvePool.sol";
+import {IVolt} from "../../volt/IVolt.sol";
+import {IPegStabilityModule} from "../IPegStabilityModule.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {MainnetAddresses} from "../../test/integration/fixtures/MainnetAddresses.sol";
+import {Constants} from "../../Constants.sol";
+
+contract CurveRouter is ICurveRouter {
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// @notice reference to the Volt contract used.
+    /// Router can be redeployed if Volt address changes
+    IVolt public immutable override volt;
+
+    constructor(IVolt _volt, TokenApproval[11] memory tokenApprovals) {
```

Comment:
we might want to add another method that allows a governor to call in and add an approval, this way we don't have to deploy a new router entirely if we miss an approval and want to add a token later on down the line